x-pack/plugins/security_solution/public/timelines/components/timeline/search_or_filter/index.tsx

```diff
@@ -110,19 +111,37 @@ const StatefulSearchOrFilterComponent = React.memo<Props>(
 
     const arrDataView = useMemo(() => (dataView != null ? [dataView] : []), [dataView]);
 
+    // Keep filter manager in sync with redux filters
+    useEffect(() => {
+      if (!deepEqual(filterManager.getFilters(), filters)) {
+        filterManager.setFilters(filters);
+      }
+    }, [filterManager, filters]);
+
+    // When a filter update comes in through the filter manager, update redux
+    useEffect(() => {
+      const subscription = filterManager.getUpdates$().subscribe(() => {
+        const filtersWithoutDropArea = getNonDropAreaFilters(filterManager.getFilters());
+        if (!deepEqual(filtersWithoutDropArea, filters)) {
+          setFilters({
+            id: timelineId,
+            filters: filtersWithoutDropArea,
+          });
+        }
+      });
+      return () => {
+        subscription.unsubscribe();
+      };
```

Comment:
if the goal here is to run this a single time, consider the following code:

```typescript
import { take } from 'rxjs';

...

return () =>
        filterManager
          .getUpdates$()
          .pipe(take(1))
          .subscribe(() => {
            const filtersWithoutDropArea = getNonDropAreaFilters(filterManager.getFilters());
            if (!deepEqual(filtersWithoutDropArea, filters)) {
              setFilters({
                id: timelineId,
                filters: filtersWithoutDropArea,
              });
            }
          });
```

But what you have looks good!